libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/DisaggregatedAnalysisUtils.ts

```diff
@@ -82,11 +84,45 @@ export function generateOverlappingFeatureBasedCohorts(
     }
   });
 
-  return (
-    generateCohortsCartesianProduct(filters, jointDataset, globalCohort)
-      // filter the empty cohorts resulting from overlapping dimensions
-      .filter((errorCohort) => errorCohort.cohortStats.totalCohort > 0)
-  );
+  const nonZeroCohorts = generateCohortsCartesianProduct(

```

Comment:
Is the above filtering needed as we don't allow users to create cohorts which are empty?

Reply:
Yes, we are automatically generating cohorts from the features with all possible combinations.  Then we filter out the empty ones.  It was like this before. I added the logic now to group the cohorts with smaller sample sizes into one "Other" cohort.